pkg/build/apk.go

```diff
@@ -85,9 +89,31 @@ func (bc *Context) InitApkKeyring() (err error) {
 	for _, element := range keyFiles {
 		log.Printf("installing key %v", element)
 
-		data, err := os.ReadFile(element)
+		asURI := uri.New(element)
+		asURL, err := url.Parse(string(asURI))
 		if err != nil {
-			return errors.Wrap(err, "failed to read apk key")
+			return errors.Wrap(err, "failed to parse key as URI")
+		}
+
+		var data []byte
+		switch asURL.Scheme {
+		case "file":
+			data, err = os.ReadFile(element)
+			if err != nil {
+				return errors.Wrap(err, "failed to read apk key")
+			}
+		case "https":
+			resp, err := http.Get(asURL.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to fetch apk key")
+			}
```

Comment:
Go doesn't return an `error` when the status code is not `2xx`, so you should also check `resp.StatusCode` and `defer resp.Body.Close()`

Reply:
Handled